packages/app/src/systems/Chains/fuel/hooks/useFuelAccountConnection.tsx

```diff
@@ -4,16 +4,25 @@ import {
   useAccount,
   useDisconnect,
   useIsConnected,
+  useProvider,
 } from '@fuels-portal/sdk-react';
+import { Address } from 'fuels';
+import { useMemo } from 'react';
 
 import { store } from '~/store';
 
-export const useAccountConnectionFuel = () => {
+export const useFuelAccountConnection = () => {
   const fuel = useFuel();
   const { account } = useAccount();
   const { isConnected } = useIsConnected();
-  const { connect, error, isLoading } = useConnect();
+  const { connect, error, isLoading: isConnecting } = useConnect();
   const { disconnect } = useDisconnect();
+  const { provider } = useProvider();
+
+  const address = useMemo(
```

Comment:
Do we gain efficiency from this being in a useMemo?  It seems to me this could just be a variable initialization with no hooks.

Reply:
yes we do, will only process Address.fromString when `account` gets changed